### PR TITLE
Upgrade to Rails 7.1.2

### PR DESCRIPTION
**Pull request by davidgisbey**
## Description

This follows the guidance set out in the EPIC [Trello card](https://trello.com/c/PhGra6VR/1809-epic-rails-712-upgrade) for Rails 7.1.2 upgrades 

I've deployed the changes to integration and it seems to be working as fine when i take all the happy paths.

## Trello card

https://trello.com/c/MvQVHuDI/1811-upgrade-content-publisher-to-rails-712

⚠️ This repo is Continuously Deployed: make sure you [follow the guidance](https://docs.publishing.service.gov.uk/manual/development-pipeline.html#merge-your-own-pull-request) ⚠️

Follow [these steps](https://guides.rubyonrails.org/upgrading_ruby_on_rails.html) if you are doing a Rails upgrade.

### CODE REVIEW

File: config/environments/production.rb [hunk 1]
```diff
@@ -74,28 +80,23 @@
...
-    config.logger    = ActiveSupport::TaggedLogging.new(logger)
-  end
-
   # Do not dump schema after migrations.
   config.active_record.dump_schema_after_migration = false
 
   config.action_mailer.delivery_method = :notify
   config.action_mailer.notify_settings = {
-    api_key: Rails.application.secrets.notify_api_key,
+    api_key: Rails.application.credentials.notify_api_key,
```
Comment (ryanb-gds) [inline config/environments/production.rb:88]: You might find that this doesn't work and that you have to use ENV directly here, as I think this gets loaded before the initialisers. See https://github.com/alphagov/whitehall/blob/b0180e9fb01c7d5168047e457402bdbefb2bbc18/config/application.rb#L72

File: spec/lib/govspeak_document/in_app_options_spec.rb [hunk 2]
```diff
@@ -14,7 +14,7 @@
                              caption: "An optional caption",
                              credit: "An optional credit",
                              filename: "filename.png")
-      edition = build(:edition, image_revisions: [image_revision])
+      edition = create(:edition, image_revisions: [image_revision])
```
Comment (ryanb-gds) [inline spec/lib/govspeak_document/in_app_options_spec.rb:17]: This seems... odd that it should be necessary, but I'm not that surprised 🤣

**Review by ryanb-gds — CHANGES_REQUESTED** [review]
I think we'll need to tweak the notify API config to use ENV directly

File: config/environments/production.rb (hunk 1 above)
Comment (davidgisbey) [reply config/environments/production.rb]: ah right that makes sense. Thanks!

**Review by ryanb-gds — APPROVED** [review]
Thanks for sorting this out

**Pull request merged**